source/include/Waveform.h

```diff
@@ -1,31 +0,0 @@
-/**
```

Comment:
I think we might need this in the future. If there are operations on waveform that are not same as operations on simple cluster, then that should be encapsulated to a class like this (for example, Timestamp). When we need to deal with variants of waveforms, the DFIR decomposition would not help and we have to dispatch operations at runtime to do the right thing. Then, abstracting those operations to a Waveform class would be the right way to do. @spathiwa  thoughts?

Reply:
I agree. And there definitely are operations on Waveforms that are treated differently than a cluster; arithmetic operations use the waveform array and ignore the the other components, for example, as does inequality comparison with a scalar constant.

Reply:
OK. But there's hardly any code in this class, we can always easily add it back when necessary. Did you want me to leave it as it is but with no references? I am OK with either.

Reply:
It is good to leave it and feel free to leave a comment that it is not used now but will soon be used. I think we will get to use this in this cycle itself. I just don't want all operations to be spread out in future PRs. There are many nodes that don't have an error cluster and we will have to support them anyway.

Reply:
Added it back.